### PR TITLE
3 introduce new form toggle styles

### PR DESCRIPTION
**Pull request by jamesjacobs**
Addresses #3

### CODE REVIEW

File: stylesheets/_component.toggle-switch.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            letter-spacing: 32px;
+        }
+    }
+}
+
+.toggle-switch-label {
+    background-clip: padding-box;
+    background-color: color(grey, light);
+    border-color: transparent;
+    border-radius: 1.5em;
```
Comment (langy) [inline stylesheets/_component.toggle-switch.scss]: Would a percentage-based unit be better here, incase the height ever changed (for whatever reason)?

File: stylesheets/_component.toggle-switch.scss [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    color: color(white);
+    display: inline-block;
+    font-size: 2em;
+    height: 1em;
+    position: relative;
+    transition: all .4s ease;
```
Comment (langy) [inline stylesheets/_component.toggle-switch.scss]: Remove 'all' and 'ease' as these are the defaults (unless they're overriding another transition)

File: stylesheets/_component.toggle-switch.scss [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        position: absolute;
+        right: 50%;
+        top: 0;
+        transition: all .3s ease-in-out;
+        transform: translate3d(0, 0, 0);
+        z-index: 2;
+    }
+
+    &:after {
+        content: "\F00C \F00D";
```
Comment (Stanton) [inline stylesheets/_component.toggle-switch.scss]: 😀

File: stylesheets/_component.toggle-switch.scss [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        border-radius: 100%;
+        border: .125em solid transparent;
+        bottom: 0;
+        content: "";
+        left: 0;
+        position: absolute;
+        right: 50%;
+        top: 0;
+        transition: all .3s ease-in-out;
+        transform: translate3d(0, 0, 0);
```
Comment (langy) [inline stylesheets/_component.toggle-switch.scss]: Prefix with -webkit- if this is a webkit-specific hack (assuming it is?)

File: views/lexicon/tabs/forms.html.twig [hunk 5]
```diff
@@ -259,6 +259,16 @@
         }}
 
         {{
+            form.toggle_switch({
+                'class': '',
+                'label': 'Toggle',
+                'id': 'toggletest',
+                'name': '',
+                'checked': false
```
Comment (langy) [inline views/lexicon/tabs/forms.html.twig]: Move this on up to the top for alphabeticallyness

File: views/pulsar/v2/helpers/form.html.twig [hunk 6]
```diff
@@ -225,6 +225,79 @@ data-*          | string | Data attributes, eg: `'data-foo': 'bar'`
...
+indeterminate | bool   | Shows the checkbox as [-], overrides the value of `checked`
+name          | string | The name of this control
+required      | bool   | Adds `required` and `aria-required="true"` attributes
+value         | string | Specifies the value of the input
+data-*        | string | Data attributes, eg: `'data-foo': 'bar'`
+
+*Any other options not listed here will be applied to the input.
+
+#}
+{% macro toggle_switch(options = {}) %}
```
Comment (Stanton) [inline views/pulsar/v2/helpers/form.html.twig]: Remove '= {}'

File: stylesheets/_component.toggle-switch.scss (hunk 4 above)
Comment (Stanton) [inline stylesheets/_component.toggle-switch.scss]: Autoprefixer 😎

Comment (jamesjacobs) [inline stylesheets/_component.toggle-switch.scss]: It's not webkit specific. It enables hardware acceleration for the animation.

File: views/lexicon/tabs/forms.html.twig [hunk 7]
```diff
@@ -259,6 +259,16 @@
         }}
 
         {{
+            form.toggle_switch({
+                'class': '',
```
Comment (Stanton) [inline views/lexicon/tabs/forms.html.twig]: Do you need the empty values here?

File: stylesheets/_component.toggle-switch.scss (hunk 4 above)
Comment (langy) [inline stylesheets/_component.toggle-switch.scss]: So serve it to all other vendors who don't need a hack to perform smoothly?

File: views/lexicon/tabs/forms.html.twig [hunk 8]
```diff
@@ -259,6 +259,16 @@
...
+                'label': 'Toggle',
+                'id': 'toggletest',
+                'name': '',
```
Comment (Stanton) [inline views/lexicon/tabs/forms.html.twig]: Is this to force a name attribute?

File: views/pulsar/v2/helpers/form.html.twig [hunk 9]
```diff
@@ -225,6 +225,79 @@ data-*          | string | Data attributes, eg: `'data-foo': 'bar'`
...
+help          | string | Additional guidance information to be displayed next to the input
+label         | string | Text for the `<label>` companion element
+required      | bool   | Visually indicates that the field must be completed
+
+## Options applied to input
+
+Option        | Type   | Description
+------------- | ------ | -------------------------------------------------------
+checked       | bool   | Whether the input is checked
+form          | string | Specific one or more forms this label belongs to
```
Comment (Stanton) [inline views/pulsar/v2/helpers/form.html.twig]: 'Specify'

File: stylesheets/_component.toggle-switch.scss (hunk 1 above)
Comment (jamesjacobs) [inline stylesheets/_component.toggle-switch.scss]: A percentage based radius doesn't create the same curve unfortunately. There is a --large modifier should you need a bigger toggle.

**Pull request merged**